org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/EventDispatcher.java

```diff
@@ -155,11 +155,13 @@ private void dispatch(EventSet eventSet) {
 				}
 				vote = true;
 				try {
-					resume = listener.handleEvent(event, fTarget, !resume, eventSet) && resume;
+					try {
+						resume = listener.handleEvent(event, fTarget, !resume, eventSet) && resume;
+					} finally {
+						enableGCForExceptionEvent(event);
+					}
 				} catch (Throwable t) {
 					logHandleEventError(listener, event, t);
-				} finally {
-					enableGCForExceptionEvent(event);
```

Comment:
I guess we can leave this in the old `finally` block?

Reply:
Sorry, I don't know what you mean. leave what and where?

Reply:
Since we are catching all exceptions in the enable GC method, we don't have to add an extra `finally` block. I thought the exception was being thrown by the code, not just logged.